sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -131,18 +145,98 @@ protected void beforeTest() {
     protected void afterTest() {
     }
 
+    /**
+     * Get offset which determines the rolling strategy.
+     *
+     * @return The offset according to day of the week and platform information
+     */
+    private static int getOffset() {
+        LocalDate today = LocalDate.now();
+        buildCalendarMap();
+        buildPlatformList();
+        return calendarMap.get(today.getDayOfWeek()) + getPlatFormOffset();
+    }
+
+    private static Integer getPlatFormOffset() {
+        String currentOs = System.getProperty("os.name").toLowerCase(Locale.ROOT);;
+        String currentJdk = System.getProperty("java.version").toLowerCase(Locale.ROOT);;
+        return platformList.stream().filter(platform ->
+            currentOs.trim().toLowerCase(Locale.ROOT).contains(platform.split(",")[0].toLowerCase(Locale.ROOT))
+            && currentJdk.trim().toLowerCase(Locale.ROOT).startsWith(platform.split(",")[1]
+                .toLowerCase(Locale.ROOT))
+        ).map(platformList::indexOf).findFirst().orElse(null);
+    }
+
+    private static void buildPlatformList() {
+        if (platformList != null) {
+            return;
+        }
+        platformList = new ArrayList<>();
```

Comment:
There's a race condition here. Instead, call this method when you declare the static variables. Same for `buildCalendarMap()`